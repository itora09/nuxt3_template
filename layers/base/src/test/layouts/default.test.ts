import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import DefaultLayout from '../../layouts/default.vue'

describe('DefaultLayout', () => {
  it('snapshot', () => {
    const wrapper = mount(DefaultLayout)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('コンポーネントがマウントされるべき', () => {
    const wrapper = mount(DefaultLayout)
    expect(wrapper.exists()).toBe(true)
  })

  it('layoutsクラスを持つべき', () => {
    const wrapper = mount(DefaultLayout)
    expect(wrapper.classes()).toContain('layouts')
  })

  it('slotコンテンツを表示するべき', () => {
    const slotContent = 'Test Content'
    const wrapper = mount(DefaultLayout, {
      slots: {
        default: slotContent,
      },
    })
    expect(wrapper.text()).toContain(slotContent)
  })
})
