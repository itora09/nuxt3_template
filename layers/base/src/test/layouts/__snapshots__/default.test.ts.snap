// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DefaultLayout > snapshot 1`] = `
<div
  class="layouts"
>
  
  
</div>
`;
